File: SourceCode_Kakao/크레인 인형뽑기 게임.java
import java.util.Vector;
class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        int boardLen = board[0].length - 1;
        Vector<Integer> tempV = new Vector<Integer>();
        
        for(int i=0; i<moves.length; ++i) {
            int tempDoll = 0;
            int pos = 0;
            for(int j=boardLen; j>=0; --j){
                if(board[j][moves[i] - 1] == 0)
                    break;
                pos = j;
                tempDoll =board[j][moves[i] - 1];
            }
            
            if(tempDoll != 0) {
                board[pos][moves[i] - 1] = 0;
                
                if(tempV.isEmpty()) {
                    tempV.addElement(tempDoll);
                }else {
                    int len = tempV.size() - 1;
                    if(tempV.elementAt(len) == tempDoll) {
                        ++answer;
                        tempV.removeElementAt(len);
                    }else {
                        tempV.addElement(tempDoll);
                    }
                }
            }
        }
        return answer * 2;
    }
}

File: history/SourceCode_Kakao/크레인 인형뽑기 게임.js
function solution(board, moves) {
    var answer = 0;
    let doll_array = [];
    let board_len = board.length - 1;
    
    moves.forEach((move_order) => {
        let temp_doll = 0;
        let pos = 0;
        for(let i = board_len;i>=0; --i) {
            if(board[i][move_order - 1]) { 
                temp_doll = board[i][move_order - 1];
                pos = i;
            }
            else 
                break;
        }
        if(temp_doll) {
            board[pos][move_order - 1] = 0;
            
            let temp_len = doll_array.length;
        
            if(temp_len) {
                if(doll_array[temp_len - 1] === temp_doll) {
                    ++answer;
                    doll_array.pop();
                }
                else
                    doll_array.push(temp_doll);
            }
            else
                doll_array.push(temp_doll);
        }
    });
    return answer * 2;
}